rect/bvhstack.go

```diff
@@ -289,6 +291,25 @@ func (s *orthStack) Score() int32 {
 	return score
 }
 
+func (s *orthStack) SAH(cInternal, cLeaves, cOverlap float64) float64 {
+	s.Reset()
+
+	var ci, cl, co float64
+
+	for s.HasNext() {
```

Comment:
I'm not sure if this is iterating through nodes multiple times -- can you confirm this is the correct way to iterate though every node once starting from root?

Reply:
This should not iterate through nodes multiple times. This is correct and similar to the "func (bvh *BVol) String()" method.